pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -1712,40 +1717,62 @@ def _send_release_email(self, release_name: str, advisories: Dict[str, int], jir
-            return
-
         candidate_pr_note = ""
         justifications = release_info["justifications"]
         if justifications:
             candidate_pr_note = "\n".join(justifications)
 
         from_releases = [arch_info["from_release"].split(":")[-1]
                          for arch_info in release_info["content"].values() if "from_release" in arch_info]
-
         advisory_id = 0
         if "advisory" in release_info and release_info["advisory"]:
             advisory_id = release_info["advisory"]
 
-        jenkins.start_cincinnati_prs(
-            from_releases,
-            release_info["name"],
-            advisory_id,
-            candidate_pr_note,
-            self.skip_ota_notification
-        )
+        release_name = release_info["name"]
+        branchName = f"pr_candidate_{release_name}"
+        pr_title = f"Enable {release_name} in candidate channel"
+        extraSlackComment = ""
+        if advisory_id != 0:
+            internal_errata_url = f"https://errata.devel.redhat.com/advisory/{advisory_id}"
+            pr_messages = f"Please merge immediately. This PR does not need to wait for an advisory to ship, but the associated advisory is {internal_errata_url} ." + candidate_pr_note
+            extraSlackComment = "automatically approved"
+        elif assembly_type in [AssemblyTypes.PREVIEW, AssemblyTypes.CANDIDATE]:
+            pr_messages = "This is a release candidate. There is no advisory associated. \nPlease merge immediately." + candidate_pr_note
+        else:
+            pr_messages = "Promoting a hotfix release (e.g. for a single customer). There is no advisory associated. \nPlease merge immediately."
+
+        # create a forked branch
+        github_client = Github(os.environ.get("GITHUB_TOKEN"))
+        upstream_repo = github_client.get_repo("openshift/cincinnati-graph-data")
+        for branch in upstream_repo.get_branches():
+            if branch.name == branchName:
+                upstream_repo.get_git_ref(f"heads/{branchName}").delete()
+        fork_branch = upstream_repo.create_git_ref(f"refs/heads/{branchName}", upstream_repo.get_branch("master").commit.sha)
+        self._logger.info("Created fork branch ref %s", fork_branch.ref)
+        # edit channel file
+        candidate_content = upstream_repo.get_contents("internal-channels/candidate.yaml", ref=branchName)
+        file_content = candidate_content.decoded_content.decode("utf-8") + f"\n- {release_name}"
+        upstream_repo.update_file("internal-channels/candidate.yaml", pr_title, file_content, candidate_content.sha, branch=branchName)
+        try:
+            pr = upstream_repo.create_pull(title=pr_title, body=pr_messages, base="master", head=branchName)
+            pr.add_to_labels("lgtm", "approved")
+            self._logger.info(f"Cincinnati PR {pr.html_url} created")
+        except GithubException as e:
+            self._logger.warning(f"Failed to update upstream repo: {e}")
```

Comment:
I think failing to create a cincy PR should cause an error.

Reply:
+1

Reply:
updated in https://github.com/openshift-eng/art-tools/pull/904/commits/b372b55eed8d23cf96affae17e7f94616d06f749